jupyter_console/interactiveshell.py

```diff
@@ -109,12 +109,23 @@ class ZMQTerminalInteractiveShell(TerminalInteractiveShell):
         """
     )
 
+    use_kernel_is_complete = Bool(True, config=True,
+        help="""Whether to use the kernel's is_complete message
+        handling. If False, then the frontend will use its
+        own is_complete handler.
+        """
+    )
+
     manager = Instance('jupyter_client.KernelManager', allow_none=True)
     client = Instance('jupyter_client.KernelClient', allow_none=True)
     def _client_changed(self, name, old, new):
         self.session_id = new.session.session
     session_id = Unicode()
 
+    def __init__(self, *args, **kwargs):
+        super(TerminalInteractiveShell, self).__init__(*args, **kwargs)
```

Comment:
`TerminalInteractiveShell` -> `ZMQTerminalInteractiveShell`. The first argument to super() should be the name of the class using it, not the parent class.